### DISCUSSION TITLE
Have you considered hosting your own Weblate for the Theme Languages?

### DISCUSSION THREAD

**Discussion by SecularSteve** [stepmania/stepmania#2195] [Ideas]
Before you ask, it's as free as it gets.
It would make the act of translating the software much easier for newer users. Let alone for languages still untouched.
And yes, it supports the .INI format.

**Comment by moruzerinho6** [comment]
The current translation system for stepmania is outdated and does not follow any modern standard, [most tools for modern sm-engine translation are being made by TIny-Foxes](https://outfox.wiki/dev/translation/#tools-and-practices).

But as noted on [fini README](https://github.com/Tiny-Foxes/fini/blob/main/README.md):

> ⚠️ StepMania Engine doesn't have proper support for more than 2 letters language files: Both StepMania and OutFox, as the date of writting this document, do not follow ISO-1 and ISO 3166-1 standards

Issues:

- While Weblate would probably help with languages that are 2 letters, it would not solve the actual problems stepmania engine has towards such topic. (Not to mention Weblate is a paid service and stepmania is a open-source no profit project, OutFox would be more likely to get it)
- There's no way to enable translations to translate images, impacting translation design.
- RTL languages are not supported.
- An absurd amount of fonts are missing if this aims to support more languages.


If you want to see community recent efforts into translation:

- OutFox Wiki - [Translation Page](https://outfox.wiki/dev/translation/) <- Documentation and good practices
- Tiny-Foxes - [OutFox-Translations](https://github.com/Tiny-Foxes/OutFox-Translations) <- Hosts all OutFox translations
- Tiny-Foxes - [Translation Tookit](https://github.com/Tiny-Foxes/Translation-Toolkit) <- Helps finding missing strings.
- Tiny-Foxes - [Stepmania-TranslatorViewMaker](https://github.com/Tiny-Foxes/Stepmania-TranslatorViewMaker) <- Makes it easier to find where to translate a line while in game
- Tiny-Foxes - [fini](https://github.com/Tiny-Foxes/Tiny-Webini) <- Brings RTL support and makes a file for each INI section of the main file.
